server/routes/induction/create/reasonsNotToGetWorkCreateController.ts

```diff
@@ -0,0 +1,55 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { ReasonsNotToGetWorkForm } from 'inductionForms'
+import ReasonsNotToGetWorkController from '../common/reasonsNotToGetWorkController'
+import validateReasonsNotToGetWorkForm from '../../validators/induction/reasonsNotToGetWorkFormValidator'
+import { getPreviousPage } from '../../pageFlowHistory'
+import getDynamicBackLinkAriaText from '../dynamicAriaTextResolver'
+import { asArray } from '../../../utils/utils'
+
+/**
+ * Controller for Reasons Not To Get Work after release screen of the create induction journey.
+ */
+export default class ReasonsNotToGetWorkCreateController extends ReasonsNotToGetWorkController {
+  getBackLinkUrl(req: Request): string {
```

Comment:
Ah, this is right/appropriate for the Update journey because there might not be a pageFlowHistory on the session, but for the Create journey there is a better/neater way of doing this 👍 
You could change this method here to:
```
  getBackLinkUrl(req: Request): string {
    const { pageFlowHistory } = req.session
    return getPreviousPage(pageFlowHistory)
  }
```

And then in the superclass (`ReasonsNotToGetWorkController`) change line 22 to:
```
if (req.session.updateInductionQuestionSet || req.session.pageFlowHistory) {
```

You'll also need to change the GET tests for this controller to provide a `pageFlowHistory` up to and including the _previous_ page:
```
      req.session.pageFlowHistory = {
        pageUrls: [
          `/prisoners/${prisonNumber}/create-induction/hoping-to-work-on-release`,
        ],
        currentPageIndex: 0,
      }
``` 

And then make an assertion that it's had "this" page appended to it:
```
      expect(req.session.pageFlowHistory).toEqual({
        pageUrls: [
          '/prisoners/A1234BC/create-induction/hoping-to-work-on-release',
          '/prisoners/A1234BC/create-induction/reasons-not-to-get-work',
        ],
        currentPageIndex: 1,
      })
```

Reply:
When I tried to update `getBackLinkUrl()` with your suggestion it started throwing errors since there was no page flow history. I've added some code to build a flow history on this page, but because there's no previous page this didn't fix the issue, so needed added a conditional to check if the previous page first exists, otherwise it uses a default.

I tried to build the flow history on the previous page (hoping to work on release) to fix this bug, but this then broke the long question set journey which currently seems to be building a history on the qualifications list page.

What I have now seems to work, but may not be "right" way to do it?